shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,55 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.repository.exception.ProductNotFoundException;
+import ro.msg.learning.shop.model.Product;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.repository.ProductRepository;
+
+import java.util.List;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductService(ProductRepository productRepository) {
```

Comment:
You can also use the @RequiredArgsConstructor annotation from Lombok to generate your constructor.